osv2/executor.go

```diff
@@ -26,45 +26,54 @@ func NewExecutor(client *opensearch.Client) *Executor {
 // will be returned.
 // An error can be returned if:
 //   - The request to OpenSearch fails
-//   - The results json cannot be unmarshalled
-func (e *Executor) MGet(ctx context.Context, req *opensearchtools.MGetRequest) (*opensearchtools.OpenSearchResponse[opensearchtools.MGetResponse], error) {
-	osv2Req := fromDomainMGetRequest(req)
-	validationRes := osv2Req.Validate()
-	if validationRes.IsFatal() {
-		return nil, opensearchtools.NewValidationError(validationRes)
+//   - The results JSON cannot be unmarshalled
+func (e *Executor) MGet(ctx context.Context, req *opensearchtools.MGetRequest) (resp opensearchtools.OpenSearchResponse[opensearchtools.MGetResponse], err error) {
+	validationResults := opensearchtools.NewValidationResults()
+
+	osv2Req, vrs := fromDomainMGetRequest(req)
+	if vrs.IsFatal() {
+		return resp, opensearchtools.NewValidationError(vrs)
```

Comment:
do we want to add the vrs to the resp incase there are any non-fatal validation results?

Reply:
Yes, good call. Done.